### DISCUSSION TITLE
I changed my UID and now I can't commit.

### DISCUSSION THREAD

**Discussion by mmance** [magit/magit#4713] [Support] [answered]
I was UID 1000 and for file share reasons I had to bump to 1001.

When I got to commit in magit:
files--ensure-directory: Creating directory: Permission denied, /run/user/1000

I am not sure where it is storing my old UID.

**Comment by kyleam** [comment]
This sounds like a general file permissions issue.  (For example, presumably you also are unable to commit from the command line.)  Have you tried changing the files so that they are owned by your new UID?

Also note that more recent Git versions will refuse to run in repositories owned by someone else (see `safe.directory` in `man git-config`).

> I am not sure where it is storing my old UID.

Magit is not storing your UID.

**Reply by mmance** [reply]
Sorry, that would have been helpful to have included that.  I can commit from cli with git without a problem.  Must be something stored in emacs.d then.  I could commit from magit before I changed my UID.  Thank you, at least I can rule that out.

**Comment by mmance** [comment]
Just for anyone else that may have this same issue. I am using spacemacs.  
In the .spacemacs.env the var for XDG_RUNTIME_DIR was hardcoded to my UID.
I don't know if magit would have hit that var, but it only did when I tried to commit.

I set the var to /run/user/$uid and problem solved.  Thanks again for magit, its pretty awesome.

**Reply by kyleam** [reply, marked as answer]
Great, thank you for following up with what you found.

> In the .spacemacs.env the var for XDG_RUNTIME_DIR was hardcoded to my UID.
> I don't know if magit would have hit that var, but it only did when I tried to commit.

Ah, I think the reason you're seeing this is that committing goes through with-editor.el, which in turns goes through server.el, and that's where this comes into play:

```elisp
;; We do not use `temporary-file-directory' here, because emacsclient
;; does not read the init file.
(defvar server-socket-dir
  (if internal--daemon-sockname
      (file-name-directory internal--daemon-sockname)
    (and (featurep 'make-network-process '(:family local))
	 (let ((runtime-dir (getenv "XDG_RUNTIME_DIR")))
	   (if runtime-dir
	       (expand-file-name "emacs" runtime-dir)
	     (expand-file-name (format "emacs%d" (user-uid))
                               (or (getenv "TMPDIR") "/tmp"))))))
  "The directory in which to place the server socket.
If local sockets are not supported, this is nil.")
```

I should have looked a bit more closely at the directory in your output, and perhaps that `/run/user/...` would have got me thinking in the right direction.
